.github/workflows/ci.yaml

```diff
@@ -0,0 +1,514 @@
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        if: |
+          github.event_name != 'pull_request' ||
+          contains(github.event.pull_request.labels.*.name, 'generate-dumps-on-pr')
+        with:
+          name: db-dump
+          path: /tmp/postgres
+
+      - name: Get db dump
+        run: |
+          source ./scripts/ci/lib.sh
+          get_db_dump
+
+      - name: Make db bundle
+        run: image/db/rhel/create-bundle.sh image/db image/db/rhel
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: scanner-db-bundle
+          path: image/db/rhel
+
+  build-images:
+    env:
+      QUAY_RHACS_ENG_RO_USERNAME: ${{ secrets.QUAY_RHACS_ENG_RO_USERNAME }}
+      QUAY_RHACS_ENG_RO_PASSWORD: ${{ secrets.QUAY_RHACS_ENG_RO_PASSWORD }}
+      QUAY_RHACS_ENG_RW_USERNAME: ${{ secrets.QUAY_RHACS_ENG_RW_USERNAME }}
+      QUAY_RHACS_ENG_RW_PASSWORD: ${{ secrets.QUAY_RHACS_ENG_RW_PASSWORD }}
+      QUAY_STACKROX_IO_RW_USERNAME: ${{ secrets.QUAY_STACKROX_IO_RW_USERNAME }}
+      QUAY_STACKROX_IO_RW_PASSWORD: ${{ secrets.QUAY_STACKROX_IO_RW_PASSWORD }}
+    runs-on: ubuntu-latest
+    needs:
+      - generate-scanner-bundle
+      - generate-scanner-db-bundle
+    if: always() && !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled')
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: scanner-bundle
+
+      - name: Unpack bundle
+        run: |
+          tar xvzf bundle.tgz
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: scanner-db-bundle
+          path: image/db/rhel
+
+      - name: Build scanner image
+        run: |
+          docker build -t scanner:"$(make --quiet tag)" -f image/scanner/rhel/Dockerfile image/scanner/rhel
+
+      - name: Build scanner-slim image
+        run: |
+            docker build -t scanner-slim:"$(make --quiet tag)" -f image/scanner/rhel/Dockerfile.slim image/scanner/rhel
+
+      - name: Build scanner-db image
+        run: |
+          docker build -t scanner-db:"$(make --quiet tag)" -f image/db/rhel/Dockerfile image/db/rhel
+
+      - name: Build scanner-db-slim image
+        run: |
+          docker build -t scanner-db-slim:"$(make --quiet tag)" -f image/db/rhel/Dockerfile.slim image/db/rhel
+
+      - name: Docker login
+        # Skip for external contributions.
+        if: |
+          github.event_name == 'push' || !github.event.pull_request.head.repo.fork
+        run: |
+          docker login -u "${QUAY_RHACS_ENG_RO_USERNAME}" --password-stdin quay.io <<<"${QUAY_RHACS_ENG_RO_PASSWORD}"
```

Comment:
what's the purpose of this step? Do we need the `RO` creds?

Reply:
To perform the `docker push` in the next step. I'm not sure why it's `RO`, but these steps were copied from `stackrox/stackrox`